File: 9041 - Software Instruction/lab06/all_whales.pl
#!/usr/bin/perl -w

my @whales;
my @input=<STDIN>;
my @flag;

sub findWhale 
{
	my $pod=0;
	my $number=0;
	my @arg=@_;
	my $whale=$arg[0];

	foreach my $string (@whales)
	{
	    if ($string =~ /:$whale/) {
	                my($count, $name) = split /:/, $string;
			$pod++;
			$number = $number + $count;
		}
	}
	$whale=~ s/\n//;
	print "$whale observations: $pod pods, $number individuals\n";
}

foreach $string (@input)
{
	$string = lc($string);
	$string =~ s/s$//;
	$string =~ s/\s+$//;
	$string =~ s/(\s)\s*/$1/g;
	$string =~ s/ /:/;
	push @whales,$string;
	my ($number, $name) = split /:/,$string;
	if ($name ~~ @flag){
		next;
	} else {
		push @flag,$name;
	}
}

@sortFlag = sort(@flag);

foreach $f (@sortFlag)
{
	$f=~ s/\n//;
	findWhale($f);
}

File: 9041 - Software Instruction/lab06/all_whales.py
#!/usr/bin/python

import sys,re

whales=[]
flag=set()

def findWhale(whale):
	pod = 0
	number = 0
	for line in whales:
		if re.search(":"+whale,line):
			strList = re.split(r":", line)
			pod = pod + 1
			number = number + int(strList[0])
	print whale + " observations: " + str(pod) + " pods, " + str(number) +" individuals"

for line in sys.stdin.readlines():
	line = line.lower()
	line = re.sub('s$','',line)
	line = re.sub('\s+',' ',line).strip()
	line = re.sub(' ',':',line, 1)
	whales.append(line)
	strList = re.split(r':',line)
	flag.add(strList[1])


flag = sorted(flag)

for f in flag:
#	f = re.sub('\n','',f)
	findWhale(f)
